packages/stream-tests/src/__tests__/deterministic-model-instance-document.test.ts

```diff
@@ -130,6 +154,60 @@ describe('ModelInstanceDocument API http-client tests', () => {
     expect(doc2.id.toString()).toEqual(doc1.id.toString())
     expect(doc2.content).toEqual(doc1.content)
   })
+
+  describe('Using custom unique metadata values', () => {
+    test('Cannot replace unique content fields values', async () => {
+      const model = await Model.create(ceramic, MODEL_DEFINITION_SET)
+
+      const doc = await ModelInstanceDocument.single(ceramic, {
+        model: model.id,
+        unique: ['foo'],
+      })
+
+      await expect(() => doc.replace({ unique: 'test' })).rejects.toThrow(
+        /Unique content fields value does not match metadata/
+      )
+
+      await doc.replace({ unique: 'foo', other: 'bar' })
+      expect(doc.content).toEqual({ unique: 'foo', other: 'bar' })
+    })
+
+    test('Cannot set content that does not pass validation', async () => {
+      const model = await Model.create(ceramic, MODEL_DEFINITION_SET)
+
+      const doc = await ModelInstanceDocument.single(ceramic, {
+        model: model.id,
+        unique: ['a'],
+      })
+      await expect(() => doc.replace({ unique: 'a', other: 'test' })).rejects.toThrow(
+        /Validation Error: data\/unique must NOT have fewer than 3 characters/
+      )
+    })
+
+    test('Can create, load and update unique documents', async () => {
+      const model = await Model.create(ceramic, MODEL_DEFINITION_SET)
+
+      const doc1 = await ModelInstanceDocument.single(ceramic, {
```

Comment:
hmm, I don't like that this function is called `single` now that it's being used for more than just the SINGLE account relation.  Can we add a different function, `set`, which always takes a `unique` field and remove the optional `unique` arg from the `single` function?